torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -57,14 +57,27 @@ def __init__(self, x):
         self.x = x
 
     def __eq__(self, other):
-        return self.__dict__ == other.__dict__
+        def eq(value, other):
```

Comment:
Probably better to add a comment on `Foo` class, mentioning that x can be either a tensor or an object.